File: game_server/ClientProtocol/Common/ClothesEnum/EAvatarGachaGuaranteeCondType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using ALBasicProtocolPack;
using UnityEngine;


namespace Common.ClothesEnum
{

/// <summary>
/// avatar抽卡条件类型
/// </summary>
public enum EAvatarGachaGuaranteeCondType {
	NONE, //0 ==== 
	[InspectorName("QUALITY - idx[1] - 筛选道具品质 QUALITY:min:max")]
	QUALITY, //1 ==== 筛选道具品质 QUALITY:min:max
	[InspectorName("TAG - idx[2] - 筛选道具TAG TAG:EXAMPLE")]
	TAG, //2 ==== 筛选道具TAG TAG:EXAMPLE
	[InspectorName("DONT_HAVE - idx[3] - 筛选还没获得的道具")]
	DONT_HAVE, //3 ==== 筛选还没获得的道具
}

public class EAvatarGachaGuaranteeCondTypeComparer : IEqualityComparer<EAvatarGachaGuaranteeCondType>{
	public bool Equals(EAvatarGachaGuaranteeCondType x, EAvatarGachaGuaranteeCondType y) { return x == y; }
	public int GetHashCode(EAvatarGachaGuaranteeCondType obj) { return obj.GetHashCode(); }
	public static int g_iEnumCount = 4;
}
}

File: game_server/ServerProtocol/src/Common/ClothesEnum/EAvatarGachaGuaranteeCondType.java
package Common.ClothesEnum;

/*********
 * avatar抽卡条件类型
 **/
public enum EAvatarGachaGuaranteeCondType {
	NONE, //0 ==== 
	QUALITY, //1 ==== 筛选道具品质 QUALITY:min:max
	TAG, //2 ==== 筛选道具TAG TAG:EXAMPLE
	DONT_HAVE, //3 ==== 筛选还没获得的道具
	;
public static final EAvatarGachaGuaranteeCondType[]  EAvatarGachaGuaranteeCondType_Values = EAvatarGachaGuaranteeCondType.values();
public static final int EAvatarGachaGuaranteeCondType_Length = EAvatarGachaGuaranteeCondType_Values.length;
public static EAvatarGachaGuaranteeCondType EAvatarGachaGuaranteeCondType_FromInt(int _ivalue) {
	if(_ivalue < 0 || _ivalue >= EAvatarGachaGuaranteeCondType_Length){ return null; }
	return EAvatarGachaGuaranteeCondType_Values[_ivalue];
}
}

